### DISCUSSION TITLE
Migration from .NET Framework runner to the .NET tool runner

### DISCUSSION THREAD

**Discussion by nils-a** [cake-build/cake#3744] [General]
@DrewRhoades wrote in https://github.com/cake-build/cake/issues/3671#issuecomment-990360539:

> ... in order to fix it, I have to rework all my build script to be compatible with Cake.Tools 2.0.0, which appears to be non-trivial (I'm currently using the full framework runner). So I'm basically stuck without investing significant effort.

Depending on your addins/modules this should really be trivial:
* You need to have .NET SDK installed on your build machine, at a minimum .NET Core 3.1, but .NET 6 is recommended
* move the pinned cake version from `tools/packages.confg` to a `tool-manifest`:
  * `dotnet new tool-manifest`
  * `dotnet tool install cake.tool`
  * remove `tools/packages.confg` from version control
* update the boostrapper, if you have one
  * get the newest from [cake-build/resources](https://github.com/cake-build/resources/tree/master/dotnet-tool) and re-apply any modifications you made.
 
This should be all there is: Two steps and probably one prerequisite installation.

Keep in mind, that the runner you're using is totally unrelated to the project you're building. So building a `net461` project using Cake 2.0.0 with the .NET tool runner (running e.g. on `net6.0`) is perfectly fine. 

Now, there can be cases where an addin/module worked in Cake 1.3.0 and does not work in Cake 2.0.0 - in this case the work to be done is lying with the addin/module author.

**Comment by DrewRhoades** [comment]
I appreciate the additional guidance.  This is actually what I attempted yesterday, before I did the work-around I described in issue #3671.  And an add-on was the initial thing I bumped into, specifically with Cake.Git 1.1.0.  It doesn't appear to be ready for Cake.Tool 2.0.0 yet.  I'll keep an eye on it and try to upgrade again later.

**Reply by nils-a** [reply]
@DrewRhoades [Cake.Git 2.0.0](https://www.nuget.org/packages/Cake.Git/2.0.0) has been released!  🎉
